src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java

```diff
@@ -813,19 +815,112 @@ void shouldPropagateErrorCodesWhileGetState() throws JsonProcessingException {
                 )
                 .expectErrorMatches(
                         e -> e instanceof NpgResponseException npgResponseException
-                                && npgResponseException.getErrors().equals(List.of(NpgClient.GatewayError.GW0001))
+                                && npgResponseException.getErrors().equals(List.of("GW0001"))
                                 && npgResponseException.getStatusCode().get().equals(HttpStatus.BAD_REQUEST)
                 )
                 .verify();
     }
 
-    private static ClientErrorDto npgClientErrorResponse(NpgClient.GatewayError gatewayError) {
+    @Test
+    void shouldPropagateErrorForUnparsableErrorResponse() {
+        UUID correlationUUID = UUID.randomUUID();
+
+        Mockito.when(paymentServicesApi.getApiClient()).thenReturn(apiClient);
+        Mockito.doNothing().when(apiClient).setApiKey(nullable(String.class));
+
+        Mockito.when(
+                paymentServicesApi.pspApiV1BuildStateGet(
+                        correlationUUID,
+                        SESSION_ID
+                )
+        )
+                .thenReturn(
+                        Mono.error(
+                                new WebClientResponseException(
+                                        "Error while invoke method for get state",
+                                        HttpStatus.BAD_REQUEST.value(),
+                                        HttpStatus.BAD_REQUEST.getReasonPhrase(),
+                                        null,
+                                        "error".getBytes(StandardCharsets.UTF_8),
+                                        null
+                                )
+                        )
+                );
+
+        StepVerifier
+                .create(
+                        npgClient.getState(
+                                correlationUUID,
+                                SESSION_ID,
+                                MOCKED_API_KEY
+                        )
+                )
+                .expectErrorMatches(
+                        e -> {
+                            if (e instanceof NpgResponseException npgResponseException) {
+                                assertTrue(npgResponseException.getErrors().isEmpty());
+                                assertEquals(HttpStatus.BAD_REQUEST, npgResponseException.getStatusCode().get());
+                                return true;
+                            } else {
+                                fail("Unexpected exception received %s".formatted(e));
+                                return false;
+                            }
```

Comment:
Are there issues with casting directly to `NpgResponseException`?

Reply:
No I just wanted here to make test failure with a custom message, I will refactor it to cast it directly

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/178/commits/000cab563e84831ac0eef2ca29fde9df363f7095